deposit/datacite/deposit.ts

```diff
@@ -0,0 +1,71 @@
+import { renderXml } from '@pubpub/deposit-utils/datacite';
+import { Resource } from 'deposit/resource';
+import { getCommunityDepositTarget } from 'server/depositTarget/queries';
+import { persistCrossrefDepositRecord, persistDoiData } from 'server/doi/queries';
+import { Collection, DefinitelyHas, Pub } from 'types';
+import { expect } from 'utils/assert';
+import { createDataciteDoiWithMetadata, createDeposit, updateDataciteDoiMetadata } from './index';
+
+type Scope = DefinitelyHas<Collection | Pub, 'community'>;
+
+export function loadScopeResourceMeta(scope: Scope, scopeResource: Resource, scopeDoi: string) {
+	scopeResource.identifiers.push({
+		identifierKind: 'DOI',
+		identifierValue: scopeDoi,
+	});
+	scopeResource.meta['created-date'] = scope.createdAt.toString();
+	if (scope.updatedAt !== scope.createdAt) {
+		scopeResource.meta['updated-date'] = scope.updatedAt.toString();
+	}
+	scopeResource.meta.publisher = scope.community.publishAs || 'PubPub';
+}
+
+const R_MISSING_ELEMENT =
+	/Missing child element.*\{http:\/\/datacite.org\/schema\/kernel-4\}(\w+).*\)./;
+const dataciteEntityToPubPubNoun = {
+	creator: 'byline contributor',
+};
+
+export async function prepareResource(scope: Scope, scopeResource: Resource, scopeDoi: string) {
```

Comment:
This function takes a resource (intermediate format of a Pub or Collection), annotates it with common metadata like its DOI, publisher name, and useful dates, and generates an XML AST and serializes it to an XML string. The XML string will ultimately be sent to the Datacite API in a DOI request, and the AST is used to render the deposit preview in PubPub.

Reply:
Good idea!